### PR TITLE
[Fix] Fix Division-by-zero error when warmup steps set the same as total steps

### PR DESCRIPTION
**Pull request by nil0x9**
Currently if user set warmup steps to be the same as total steps (which is deemed valid) for `cosine` or `linear` LR scheduler, a `division-by-zero` error will be thrown. This PR fixes this case (along w/ some lint issues).

### CODE REVIEW

File: xtuner/v1/train/trainer.py [hunk 1]
```diff
@@ -764,7 +764,7 @@ def build_engine(
             engine.model.set_hf(model_path)
         return engine
 
-    def build_lr_scheduler(self, lr_cfg: LRConfig, scheduler_step: int) -> torch.optim.lr_scheduler.LRScheduler:
+    def build_lr_scheduler(self, lr_cfg: LRConfig, scheduler_steps: int) -> LRScheduler:
```
Comment (HAOCHENYE) [inline xtuner/v1/train/trainer.py]: Keep the signatiure of `build_lr_scheduler`
